src/keyDerivation.c

```diff
@@ -0,0 +1,75 @@
+#include <os_io_seproxyhal.h>
+#include <stdint.h>
+
+#include "assert.h"
+#include "errors.h"
+#include "keyDerivation.h"
+
+#define VALIDATE_PARAM(cond) if (!(cond)) THROW(ERR_INVALID_REQUEST_PARAMETERS)
+
+void validatePath(uint32_t* path, uint32_t pathLength)
+{
+	uint32_t bip44 = BIP_44 | HARDENED_BIP32;
+	uint32_t adaCoinType = ADA_COIN_TYPE | HARDENED_BIP32;
+
+	VALIDATE_PARAM(pathLength >= 3 && pathLength <= 10);
+
+	VALIDATE_PARAM(path[0] ==  bip44);
+	VALIDATE_PARAM(path[1] ==  adaCoinType);
+	VALIDATE_PARAM(path[2] >= HARDENED_BIP32);
+}
+
+void derivePrivateKey(
+        uint32_t* bip32Path, uint32_t pathLength,
+        uint8_t* chainCode,
```

Comment:
chain code should be made a struct so it is obvious what is the length of the code.
Also, I would typedef cx_ecfp_256_extended_private_key_t as privateKey_t, if we ever need to change it